applets/notification_area/na-grid.h

```diff
@@ -48,6 +50,7 @@ struct _NaGridClass
 };
 
 GType           na_grid_get_type        (void);
+void            set_grid_display_mode   (NaGrid *grid, gint min_icon_size);
```

Comment:
You should use `na_` prefix here... But I think you can handle what you want in `na-grid.c` and this is not even needed.

Reply:
I tried to set a size-allocate function on the grid, that was the first thing I did (i added it to the GtkGridClass, it didn't work to update the grid. I'm not an expert in Gtk.. Is there a difference between setting it as a g_signal_callback and on the Class virtual functions? If I knew how to make it work I would not have created that function, I agree, it's not its place..

Reply:
Well you can do `g_signal_connect (self, "size-allocate", callback, data);` in na-grid.c, no? Other option is to use `widget_class->size_allocate = na_grid_size_allocate;`

Reply:
I did set it up as widget_class->size_allocate as the first option, it didn't work.. but I haven't tried g_signal_connect on the gird directly, that's why I'm asking if there's a specific difference between setting it as a virtual_method on the class, or as a signal callback? In any case I'll try again to figure out how to get rid of the size-allocate in main.c.

Reply:
Well it should work in both cases... What exactly did not work?

Reply:
Strange, I just tried as a g_signal.. it works perfectly! Yet I remember clearly I had tried as a class fucntion before and it didn't work.. Anyway, now that I know it works, I'll get on it tomorrow and all the inconsistencies you pointed out. Tonight I can't continue anymore.